blackdoc/blacken.py

```diff
@@ -0,0 +1,47 @@
+import copy
+import re
+
+import black
+from blib2to3.pgen2.tokenize import TokenError
+
+from .formats import extract_code, reformat_code
+
+
+def update_line_number(message, original_number):
+    line_re = re.compile(r"(?P<line_number>\d+):(?P<column_number>\d+):")
+    match = line_re.search(message)
+    if match:
+        line_number, column_number = map(int, match.groups())
+        new_line_number = line_number + original_number - 1
+
+        message = line_re.sub(f"{new_line_number}:{column_number}:", message)
+    return message
+
+
+def blacken(lines, mode=None):
```

Comment:
same here, but that would probably be integration tests instead?

Reply:
integration tests for these are a bit more work, so let's leave them for a different PR